src/VisualStudio/Core/Def/Implementation/Venus/ContainedDocument.cs

```diff
@@ -117,7 +117,7 @@ internal sealed class ContainedDocument : ForegroundThreadAffinitizedObject, IVi
 
             this.Key = new DocumentKey(Project, filePath);
             this.Id = DocumentId.CreateNewId(Project.Id, filePath);
-            this.Folders = containedLanguage.Project.GetFolderNames(itemId);
+            this.Folders = (containedLanguage.Project as AbstractLegacyProject)?.GetFolderNames(itemId);
```

Comment:
I believe Venus is not in plan for Dev15 for CPS, though @davkean @srivatsn can confirm. I'll change it to a hard cast.